src/rules/ng-component-decorator-props-order.ts

```diff
@@ -1,27 +1,33 @@
-import { TSESTree } from '@typescript-eslint/utils';
-import { ALL_COMPONENT_DECORATORS } from '../selectors';
 import { createEslintRule } from '../utils/create-eslint-rule';
-import { fixUnsortedProps } from "../utils/fix-unsorted-props";
-import { getArgumentsFromIdentifier } from "../utils/get-arguments-from-identifier";
-import { getPropsFromObject } from "../utils/get-props-from-object";
-import { getSortedPropsWhenUnsorted } from "../utils/get-sorted-props-when-unsorted";
-import { getSourceCodeFromAsts } from "../utils/get-source-code-from-asts";
+import { forceDecoratorPropsToBeSorted } from '../utils/force-decorator-props-to-be-sorted';
 
 export const ruleName = 'ng-component-decorator-props-order';
 export const messageId = 'ngComponentDecoratorPropsOrderError' as const;
 
+const allNgComponentsSelector = 'ClassDeclaration > Decorator > CallExpression > Identifier[name="Component"]';
 const allPropNamesSorted = [
 	'selector',
 	'exportAs',
 	'templateUrl',
 	'template',
+	'styleUrl',
 	'styleUrls',
 	'styles',
 	'animations',
 	'changeDetection',
+	'encapsulation',
+	'preserveWhitespaces',
 	'standalone',
+	'jit',
+	'inputs',
+	'outputs',
 	'imports',
 	'providers',
+	'viewProviders',
+	'queries',
+	'schemas',
+	'hostDirectives',
+	'host',
```

Comment:
We tend to put host next to template/styles, I'd put it after `styles` (because the most common use case is setting a class, which is used in the styles)